lib/src/create_report.dart

```diff
@@ -503,13 +508,62 @@ Future<ReportSection> _trustworthyDependency(
     if (pubspec.dartSdkConstraint != null &&
         pubspec.dartSdkConstraint.allows(currentSdkVersion)) {
       try {
-        final outdated = await toolEnvironment.runPubOutdated(packageDir,
-            args: ['--json', '--no-dev-dependencies']);
-        for (final package in outdated['packages'] as List) {
+        final outdated = Outdated.fromJson(await toolEnvironment.runPubOutdated(
+            packageDir,
+            args: ['--json', '--up-to-date', '--no-dev-dependencies']));
+
+        String constraint(Dependency dependency) {
+          if (dependency is HostedDependency) {
+            return dependency.version.toString();
+          } else if (dependency is SdkDependency) {
+            return dependency.sdk;
+          } else if (dependency is GitDependency) {
+            return dependency.ref;
+          } else if (dependency is PathDependency) {
+            return dependency.path;
+          } else {
+            return '-';
+          }
+        }
+
+        final table = [
+          ['Package', 'Constraint', 'Compatible', 'Latest'],
+          [':-', ':-', ':-', ':-'],
+          for (final package in outdated.packages)
+            if (pubspec.dependencies.containsKey(package.package))
+              [
+                '[${package.package}]',
+                constraint(pubspec.dependencies[package.package]),
+                package.upgradable?.version ?? '-',
+                package.latest?.version ?? '-',
+              ],
```

Comment:
nit, one could also do map... But this is also fine :D
```suggestion
...outdated.packages
           .where((p) => pubspec.dependencies.containsKey(p.package))
           .map((p) => [
  '[${p.package}]',
  constraint(pubspec.dependencies[p.package]),
  p.upgradable?.version ?? '-',
  p.latest?.version ?? '-',
]),
```

Reply:
Yeah - its approximately the same.